### ISSUE TITLE
More information on the new virtual_machine resource

### ISSUE THREAD

**Issue by sofixa** [terraform-providers/terraform-provider-vsphere#240]
Hi, @vancluever . How have you been? 

I've noticed that in #212 you mentioned the new virtual machine resource and that it will be out soon, so i have some follow up questions, most notably:

* would it be possible to get an overview of its functionalities in the v1 and the potential future roadmap(e.g. if modification of VMs isn't possible in v1, is it in the roadmap? etc. etc. etc.) ?

* do you have a relative ETA? If it's more than 2-3 weeks, would it be possible to do a private beta(we're pretty interested in using the provider on a big scale, but the uncertainty over it's feature set is kinda killing us and we've been on hold; some clarity on the subject would be **greatly** appreciated ) ?

Thanks in advance,
Cheers,
Adrian

**Comment by vancluever** [comment]
Hi @sofixa!

I'm excited to say that the PR for the VM resource should be following very soon - probably in the next couple of days, as porting the existing tests, adding a couple of more tests, and documentation are the only remaining tasks before we start the review process for shipping.

You can see the branch here: https://github.com/terraform-providers/terraform-provider-vsphere/tree/f-vm-resource-refactor

Some highlights:

* VM deployment is supported via both "bare metal" creation and deployment from a template.
* Ability to do CPU and memory hot-plug.
* Support for both host and storage vMotion.
* The coupling between customization and network devices has been removed, along with the code that was the longtime source of panics in the old resource.
* Customization has been moved exclusively to the clone workflow and won't be attempted unless you are cloning. Cloning can also be done without customization.
* New customization options have been added, exposing nearly all of the customization spec.

That's the major points I can think of off the top of my head right now. As this is a major rewrite, there is a lot of new things here, and when the dust settles and the resource is close to merging, we will do something a bit more formal.

As far as plans, the major things still on the TODO that haven't made it into this release of the resource are DRS and OVF import, in order of the complexity of implement - DRS should be simple now that we have vmotion capability, as we can pull recommendations into Terraform and then just vMotion, but OVF import will require some thought on how importing the virtual disks and extra config will work.

The 2-3 week ETA is accurate to what I hope to see over on this side, can't predict 100%, but know that this is pretty much priority number 1 for us on the provider right now.

Thanks and stay tuned!

**Comment by vancluever** [comment]
Hey anyone watching this, #244 is now in. If you are curious about what is going into the new resource and what you might want to do to start preparing for it, feel free to look the full documentation, which has been linked in the PR as well.

Thanks!

**Comment by rismoney** [comment]
Will existing 0.4.2 configurations still work or will they bork the state file and FNR...asking for a friend :)

**Comment by vancluever** [comment]
@rismoney 0.4.2 states will work, but the configuration has changed significantly, you will need to update the configuration with the new config. Your state will be migrated on the next plan/apply cycle. The path is similar to importing, without the need to import. As is good general TF practice, make sure you plan and review the plan before executing any applies.

Check the docs for more details! (The new ones are in master but should be available on the site this afternoon)
